lti_consumer/lti_xblock.py

```diff
@@ -922,6 +958,28 @@ def is_past_due(self):
             close_date = due_date
         return close_date is not None and timezone.now() > close_date
 
+    def _get_lti_advantage_deep_linking_enabled(self, consumer):
+        """
+        Return whether the LTI 1.3 Advantage Deep Linking is enabled. Use the LTI consumer to determine whether deep
+        linking is enabled instead of directly accessing the xBlock fields, the LTI Configuration model, or the
+        external configuration.
+        """
+        try:
+            deep_linking_enabled = consumer.lti_dl
+            return deep_linking_enabled is not None
```

Comment:
it can never be none because lti_dl can't return none, only an exception - lti_dl behavior is already confusing in this PR